tests/test_query_engine.py

```diff
@@ -1292,3 +1293,87 @@ class Cohort(OldCohortWithPopulation):
         dict(patient_id=1, code="abc"),
         dict(patient_id=2, code="xyz"),
     ]
+
+
+def test_dsl_code_comparisons(cohort_with_population, engine):
+    if engine.name == "spark":
+        pytest.xfail()
+
+    input_data = [
+        patient(1, ctv3_event("abc")),
+        patient(2, ctv3_event("abc")),
+        patient(3, ctv3_event("def")),
+    ]
+    engine.setup(input_data)
+
+    events = tables.clinical_events
+    first_code = (
+        events.sort_by(events.code).first_for_patient().select_column(events.code)
+    )
+
+    date_categories = {
+        "abc": first_code == "abc",
+        "not_abc": first_code != "abc",
+    }
+
+    data_definition = cohort_with_population
+    data_definition.code_group = dsl_categorise(date_categories, default="unknown")
+
+    result = engine.extract(data_definition)
+
+    assert result == [
+        {"patient_id": 1, "code_group": "abc"},
+        {"patient_id": 2, "code_group": "abc"},
+        {"patient_id": 3, "code_group": "not_abc"},
+    ]
+
+
+def test_dsl_date_comparisons(cohort_with_population, engine):
+    """
+    Exercise comparison (and some boolean) operators in the DSL
+
+    We want to ensure the PatientSeries comparison and boolean operators work
+    as expected for date values.  We're using the DSL's categorise function
+    here to let us make boolean values against which to match the PatientSeries
+    values.
+    """
+    if engine.name == "spark":
+        pytest.xfail()
+
+    input_data = [
+        patient(1, ctv3_event("abc", "2019-12-31")),
+        patient(2, ctv3_event("abc", "2020-02-29")),
+        patient(3, ctv3_event("abc", "2020-10-01")),
+        patient(4, ctv3_event("abc", "2021-04-07")),
+    ]
+    engine.setup(input_data)
+
+    events = tables.clinical_events
+    first_code_date = (
+        events.sort_by(events.date).first_for_patient().select_column(events.date)
+    )
+
+    first_half_2020 = (first_code_date >= "2020-01-01") & (
+        first_code_date <= "2020-03-31"
```

Comment:
I beg you to make this date 30th June. I will buy you actual beer if you humour me.